src/utils/generateEventLink.ts

```diff
@@ -0,0 +1,102 @@
+import {
```

Comment:
I would not say so, I think it depends on a case, but here it is clear for me it is just one function which does something one and does not keep any attributes.

Reply:
It could be an entity (e.g. EventGenerator or something like this) which is responsible for links generation, it could play a role of Facade which isolates `calendar-link` implementation nuances. After that, we could replace `calendar-link` with anything else (e.g. our own implementation of this package) and other parts of our app wouldn't care, because interface remains the same.

Here it's a procedural code now, which we would bring into other classes and mix OOP with procedural code and if anything changes here, we'd need to change it in other places, because there is no sustainable API interface by which links generation is communicated.

And it has at least one attribute: `FORMAT`, but it's not really about how many attributes it has, it may have 0.

Reply:
how come will it mix, the entry point here only one function, and it will be the same both for procedural and OOP code.
What is the point to use Classes, if we do not plan to inherit it, have several instances and other OOP features?
I think it is just a utility. Yes, it has some helpers, but it is just a matter of structuring.

>  because there is no sustainable API interface by which links generation is communicated.

How come? it will always require event and calendarType

> it could play a role of Facade which isolates calendar-link implementation nuances. After that, we could replace calendar-link with anything else (e.g. our own implementation of this package) and other parts of our app wouldn't care, because interface remains the same.

Now is different?

Sorry, I do not see any benefits here to use OOP.

Reply:
I meant to keep it in a proper (Java-like or C#-like way: classes only), but you are right, in JS it's about structure and we may use modules in the same way.